Numerous fixes to rtt_tf's tf_interface, also filling out rtt_rosparam's service requester

jbohren opened the pull request:
- **rtt_tf:** Adding `canTransform` RTT_TF operation
- **rtt_rosparam:** Adding `getParam` and `setParam` for getting and setting params which have different names (or namespaces) from their property names
- **rtt_rosdeployment:** Adding a plugin which lets you manipulate an OCC::DeploymentComponent over ROS service calls (like run scripts, list components, etc)
- **rtt_ros_msgs:** .srv types used by rtt_rosdeployment
- **rtt_rosclock:** **_significant change see new rtt_rosclock README.md**_ clarifying which operations handle wall time vs sim time

jbohren committed:
rtt_rosdeployment: Interacting with a deployer via ROS service calls

jbohren committed:
rtt_rosclock: Clarifying walltime vs simtime clock access

jbohren committed:
Merge branch 'hydro-devel' of github.com:jhu-lcsr-forks/rtt_ros_integration into hydro-devel

jbohren commented:
I've added a bunch of things into this, see the updated description and package README files for more details.

jbohren commented:
Bump

meyerj commented on rtt_rosclock/src/rtt_rosclock.cpp:
```diff
@@ -10,44 +10,66 @@ namespace rtt_rosclock {
+const ros::Time rtt_rosclock::rtt_now() 
+{
+  // NOTE: getNSecs returns wall time, getTicks returns offset time
+  uint64_t nsec64 = RTT::os::TimeService::ticks2nsecs(RTT::os::TimeService::Instance()->getTicks()); //RTT::os::TimeService::Instance()->getNSecs();
```
In `rtt_rosclock::rtt_now()` and `rtt_rosclock::rtt_wall_now()`:

I would suggest to use type `RTT::nsecs` here instead of `uint64_t`, which is the return type of `ticks2nsecs()` and resolves to `long long`. Furthermore `1E9` should be replaced by `1000000000ll` as constants in scientific notation are floating point constants, making the operations floating point instead of integer operations.

jbohren replied:
All sounds reasonable.

jbohren replied:
So upon review of this, I think I want to stick with uint64_t since it's meant to ensure that the separation below into sec and nsec happens correctly with the uint32_t. I am, however, changing `1E9` to `1000000000ll`.

meyerj commented on rtt_rosdeployment/src/rtt_rosdeployment_service.cpp:
```diff
@@ -0,0 +1,138 @@
+*     THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR 
+*     OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY 
+*     OF SUCH DAMAGE.
+**/
```
This license header looks like copied from somewhere else? According to the `package.xml` file `rtt_rosdeployment` has a BSD license?

jbohren replied:
I'll resolve that.

jbohren committed:
Fixup before merge

rtt_rosclock: using long-long literals instead of floats in time
conversion
rtt_rosdeployment: using wrong license that snuk in

jbohren commented:
Resolved @meyerj's concerns. Anything else before this gets merged?
